src/librustc/hir/check_attr.rs

```diff
@@ -53,33 +53,36 @@ impl<'a> CheckAttrVisitor<'a> {
             }
         };
         for word in words {
-            let word: &str = &word.name();
-            let message = match word {
-                "C" => {
-                    if target != Target::Struct && target != Target::Enum {
-                            "attribute should be applied to struct or enum"
-                    } else {
-                        continue
+            let name = word.name();
+            let message = match name {
+                Some(word) => match &*word {
```

Comment:
I'll use `if let ...`. The check you're referring to happens [https://github.com/rust-lang/rust/pull/35850/files#diff-21e1f9f42b7976b50c28979848df2e04R951](here).